rust/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -270,16 +272,31 @@ impl HyperlaneLogStore<MerkleTreeInsertion> for HyperlaneRocksDB {
 }
 
 #[async_trait]
-impl HyperlaneMessageStore for HyperlaneRocksDB {
-    /// Gets a message by nonce.
-    async fn retrieve_message_by_nonce(&self, nonce: u32) -> Result<Option<HyperlaneMessage>> {
-        let message = self.retrieve_message_by_nonce(nonce)?;
+impl HyperlaneSequenceIndexerStore<HyperlaneMessage> for HyperlaneRocksDB {
+    /// Gets data by its sequence.
+    async fn retrieve_by_sequence(&self, sequence: u32) -> Result<Option<HyperlaneMessage>> {
+        let message = self.retrieve_message_by_nonce(sequence)?;
         Ok(message)
     }
 
-    /// Retrieve dispatched block number by message nonce
-    async fn retrieve_dispatched_block_number(&self, nonce: u32) -> Result<Option<u64>> {
-        let number = self.retrieve_dispatched_block_number_by_nonce(&nonce)?;
+    /// Gets the block number at which the log occurred.
+    async fn retrieve_log_block_number(&self, sequence: u32) -> Result<Option<u64>> {
+        let number = self.retrieve_dispatched_block_number_by_nonce(&sequence)?;
+        Ok(number)
+    }
+}
+
+#[async_trait]
+impl HyperlaneSequenceIndexerStore<MerkleTreeInsertion> for HyperlaneRocksDB {
+    /// Gets data by its sequence.
+    async fn retrieve_by_sequence(&self, sequence: u32) -> Result<Option<MerkleTreeInsertion>> {
+        let insertion = self.retrieve_merkle_tree_insertion_by_leaf_index(&sequence)?;
+        Ok(insertion)
+    }
+
+    /// Gets the block number at which the log occurred.
+    async fn retrieve_log_block_number(&self, sequence: u32) -> Result<Option<u64>> {
+        let number = self.retrieve_merkle_tree_insertion_block_number_by_leaf_index(&sequence)?;
```

Comment:
I wasn't able to find where the `store_` counterpart of this is called, although `get_next_range` depends on this call. Wondering how it didn't cause any issues on your local, maybe because of the ism setup?

Reply:
oh good call - the reason things worked is because it's not strictly required